code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -750,23 +788,53 @@
 ////////  Internal damage  ////////
 ///////////////////////////////////
 
-/obj/vehicle/sealed/mecha/proc/check_for_internal_damage(list/possible_int_damage,ignore_threshold=null)
-	if(!islist(possible_int_damage) || !length(possible_int_damage))
+/// tries to repair any internal damage and plays fluff for it
+/obj/vehicle/sealed/mecha/proc/try_repair_int_damage(mob/user, flag_to_heal)
+	balloon_alert(user, get_int_repair_fluff_start(flag_to_heal))
+	log_message("[key_name(user)] starting internal damage repair for flag [flag_to_heal]", LOG_MECHA)
+	if(!do_after(user, 10 SECONDS, src))
+		balloon_alert(user, get_int_repair_fluff_fail(flag_to_heal))
+		log_message("Internal damage repair for flag [flag_to_heal] failed.", LOG_MECHA, color="red")
 		return
-	if(prob(20))
-		if(ignore_threshold || atom_integrity*100/max_integrity < internal_damage_threshold)
-			for(var/T in possible_int_damage)
-				if(internal_damage & T)
-					possible_int_damage -= T
-			if (length(possible_int_damage))
-				var/int_dam_flag = pick(possible_int_damage)
-				if(int_dam_flag)
-					set_internal_damage(int_dam_flag)
-	if(prob(5))
-		if(ignore_threshold || atom_integrity*100/max_integrity < internal_damage_threshold)
-			if(LAZYLEN(equipment))
-				var/obj/item/mecha_parts/mecha_equipment/ME = pick(equipment)
-				qdel(ME)
+	clear_internal_damage(flag_to_heal)
+	balloon_alert(user, get_int_repair_fluff_end(flag_to_heal))
+	log_message("Finished internal damage repair for flag [flag_to_heal]", LOG_MECHA)
+
+///gets the starting balloon alert flufftext
+/obj/vehicle/sealed/mecha/proc/get_int_repair_fluff_start(flag)
+	switch(flag)
+		if(MECHA_INT_FIRE)
+			return "Activating internal fire supression..."
+		if(MECHA_INT_TEMP_CONTROL)
+			return "Resetting temperature module..."
+		if(MECHA_INT_TANK_BREACH)
+			return "Activating tank sealant..."
+		if(MECHA_INT_CONTROL_LOST)
+			return "Recalibrating coordination system..."
+
+///gets the on-fail balloon alert flufftext
+/obj/vehicle/sealed/mecha/proc/get_int_repair_fluff_end(flag)
+	switch(flag)
```

Comment:
messed the comments with this proc and the next one

Reply:
Done